File: src/service/ModelBuilder.Models/Data/Composite.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Mb.Models.Data
{
    public class Composite
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string SemanticReference { get; set; }
        public virtual ICollection<Attribute> Attributes { get; set; }
        public virtual string NodeId { get; set; }

        [JsonIgnore]
        public Node Node { get; set; }
    }
}

File: src/client/src/models/classes/Composite.ts
import { Attribute } from "..";

export const COMPOSITE_KIND: string = "Composite";

class Composite {
  id: string;
  name: string;
  semanticReference: string;
  attributes: Attribute[];
  nodeId: string;

  kind: string = COMPOSITE_KIND;

  constructor(composite: Composite) {
    Object.assign(this, composite);
  }
}

export default Composite;
